api/common/api-profile.go

```diff
@@ -0,0 +1,457 @@
+package apicommon
```

Comment:
To discuss, I have a bit the feelign this file could be splitted into 2 pieces.
There is all the definitions of the Profile strcuture and conversions methods which are API related.
There is also methods and logic for the internal validation. We could move all of this into a module/tool in /internal. It would avoid to have for exemple the default-profile into /api whereas it is something which is used very deep by UserProfileCache system.